azerothcore/azerothcore-wotlk
# refactor(DB/Misc): Improve loading time

* NOTE: Running these SQL files can take several minutes (up to 30minutes depending on your system!)

* This will improve loading time over 50%



## Changes Proposed:
* NOTE: Running these SQL files can take several minutes (up to 30minutes depending on your system!)
* It can feel like the updater gets stuck or freezses. It doesnt. Just a lot of queries. A small price to pay one time for a big improvement in performance

On my system it took about 3-4 minutes to run it

-  Update engine, collation and get rid of MEDIUMINT in all databases
-  These changes will cut loading worldserver by over 50%

Before these changes
`WORLD: World Initialized In 0 Minutes 18 Seconds`

After these changes
`WORLD: World Initialized In 0 Minutes 7 Seconds`

## SOURCE:

Inspired by https://github.com/TrinityCore/TrinityCore/commit/c102baf99e1640df72a25444e3f206e1920a0b18
## Tests Performed:

- Tested on local machiene. Huge improvement in loading times.


## How to Test the Changes:


1. Import the changes with the autoupdater (Risk is that using i.e Heidi it will crash du to Heidi being mediocre)
2. See performance bonus 😎 


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
